Update observability doc structure and content (#35816)
ray-project/ray: doc/source/cluster/kubernetes/user-guides/logging.md

# Log Persistence

> Logs (both system and application logs) are useful for troubleshooting Ray applications and system. For example, you may want to access system logs if a node dies unexpectedly.

Reviewer: ```suggestion
Logs (both system and application logs) are useful for troubleshooting Ray applications and your system. For example, you may want to access system logs if a node terminates unexpectedly.
```

> Similar to Kubenetes, Ray does not provide a native storage solution for log data. Users need to manage the lifecycle of the logs by themselves. This page provides instructions on how to collect logs from Ray clusters running on Kubernetes.

Reviewer: ```suggestion
Similar to Kubernetes, Ray does not provide a native storage solution for log data. Users need to manage the lifecycle of the logs by themselves. This page provides instructions on how to collect logs from Ray Clusters that are running on Kubernetes.
```

:::{tip}
Skip to {ref}`the deployment instructions <kuberay-logging-tldr>`
for a sample configuration showing how to extract logs from a Ray pod.
:::

## Ray log directory

> By default, Ray writes logs to files in the directory `/tmp/ray/session_*/logs` on each Ray pod's file system, including application logs and system logs. Learn more about the {ref}`log directory and log files <logging-directory>` before you start to collect the logs.

Reviewer: ```suggestion
By default, Ray writes logs to files in the directory `/tmp/ray/session_*/logs` on each Ray pod's file system, including application and system logs. Learn more about the {ref}`log directory and log files <logging-directory>` before you start to collect the logs.
```

## Log processing tools
There are a number of open source log processing tools available within the Kubernetes ecosystem. This page will shows how to extract Ray logs using [Fluent Bit][FluentBit].
Other popular tools include [Vector][Vector], [Fluentd][Fluentd], [Filebeat][Filebeat], and [Promtail][Promtail].

## Log collection strategies

> Here lists two strategies for collecting logs written to a pod's filesystem,

Reviewer: I am taking a guess at the `container orchestration` term. Feel free to remove if it's not helpful.

> Here lists two strategies for collecting logs written to a pod's filesystem,

Author: @kevin85421 to comment on this

> Here lists two strategies for collecting logs written to a pod's filesystem,

Author: it's not really a container orchestration. The strategies here are basically the first two "cluster-level logging architectures" they list in their [documentation](https://kubernetes.io/docs/concepts/cluster-administration/logging/#cluster-level-logging-architectures).

I'll just keep it simple for now
"Collect logs written to a pod's filesystem using one of two logging strategies:"

:language: yaml
```

(kuberay-logging-tldr)=

> ### Deploying a RayCluster with logging CR

Reviewer: Do we need to spell out CR to avoid confusion?

Author: Not sure. cc: @kevin85421

Author: I believe we want to use "sidecar" instead of "CR"

> ### Deploying a RayCluster with logging CR

Author: I'm afraid RayCluster is a specific term? cc: @kevin85421

> ### Deploying a RayCluster with logging CR

Author: RayCluster is the Custom Resource (CR) name. Keep it as is.

<img width="1288" alt="Screenshot 2023-05-31 at 4 42 18 PM" src="https://github.com/ray-project/ray/assets/9677264/367439fb-cedc-4a01-8506-755187352760">

> Now, we will see how to deploy the configuration described above.

Reviewer: ```suggestion

```

> Deploy the KubeRay Operator if you haven't yet.

Reviewer: ```suggestion
To deploy the configuration described above, deploy the KubeRay Operator if you haven't yet:
```

[KubDoc]: https://kubernetes.io/docs/concepts/cluster-administration/logging/
[ConfigLink]: https://raw.githubusercontent.com/ray-project/ray/releases/2.4.0/doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml

## Redirecting Ray logs to stderr

> By default, Ray logs are written to files under the ``/tmp/ray/session_*/logs`` directory. It may not be ideal if the log processing tool needs log to be written to stderr in order for them to be captured. View {ref}`configuring logging <redirect-to-stderr>` for details on how to redirect all the logs to stderr of the host pods instead.

Reviewer: ```suggestion
By default, Ray writes logs to files under the ``/tmp/ray/session_*/logs`` directory. Capturing logs from stderr may not be ideal for log processing tools. View {ref}`configuring logging <redirect-to-stderr>` for details on how to redirect logs to the stderr of the host pods instead.
```

Author: > Capturing logs from stderr may not be ideal for log processing tools.

RE this, what we want to say here is: `Writing to log files is not ideal if users or the log processing tools prefer to capture logs from stderr.`
